### DISCUSSION TITLE
Is there a way to style darkmode emails?

### DISCUSSION THREAD

**Discussion by emilianoyaryura** [resend/react-email#591] [Q&A]
I need to style the emails to render appropriate styles in light and dark mode. The `<Tailwind />` component doesn't seems to support `dark:`

**Comment by janwagner** [comment]
I am doing it like this

1. Add meta tags to the `<Head>`
```
<meta content='light dark' name='color-scheme' />
<meta content='light dark' name='supported-color-schemes' />
```

2. Add style tag to the head
```
<style type='text/css'>
    @media (prefers-color-scheme: dark) {
        .logo.light {
            display: none !important;
          }
        .logo.dark {
            display: block !important;
        }
        body {
          background-color: #XXXXXX !important;
        }
        a[data-id*="react-email-button"] {
            background-color: #XXXXXX !important;
            color: #ffffff !important;
        }
        [data-id*="react-email-heading"],
        [data-id*="react-email-text"] {
          color: #ffffff !important;
        }
      }
</style>
```

3. Add darkmode images
```
<span className='logo light'>
    <Img
        src='https://...'
        alt='Default Logo'
        width=''
        height=''
    />
</span>
<span className='logo dark' style={{ display: 'none' }}>
    <Img
        src='https://...'
        alt='Darkmnode Logo'
        width=''
        height=''
    />
</span>
```

**Comment by JuanSpada** [comment]
Hello, you could do it? Can you show us how.

**Reply by emilianoyaryura** [reply]
Al final no lo tuve que hacer. [Acá](https://github.com/resend/react-email/discussions/591#discussioncomment-5836755) me respondieron una solución que no pude probar. También, por si te sirve, había encontrado esta [librería](https://github.com/mjmlio/mjml) que parecería soportarlo por lo que leí ([art.](https://scribe.rip/m/global-identity?redirectUrl=https%3A%2F%2Fblog.like.co%2Fmjml-react-email-dark-mode-44b4a53152bf))

**Comment by joshdoesthis** [comment]
You can use Tailwind CSS utility classes the way you might elsewhere, provided that you wrap the necessary components with the Tailwind component from `@react-email/components` (or `@react-email/tailwind`). No need to explicitly type out any CSS as it will be figured out by the Tailwind component. At least one caveat to note is that different email clients do different things, including different things with the media query for `prefers-color-scheme`. I've just faced the issue where Apple Mail on macOS was doing strange things and this is an example of the solution:

```ts
import { Body, Head, Html, Section, Tailwind, Text } from '@react-email/components'
import { render } from '@react-email/render'

const ExampleEmail: FC = () => {
  return (
    <Tailwind>
      <Html>
        <Head>
          <style>{':root {color-scheme: light dark;}'}</style> // Required for dark mode to work in Apple Mail on macOS.
        </Head>
        <Body className='p-4'>
          <Section className='bg-white dark:bg-black text-black dark:text-white'>
            <Text>Just an example email.</Text>
          </Section>
        </Body>
      </Html>
    </Tailwind>
  )
}

const html = await render(<ExampleEmail />)
console.log({ html })
const text = await render(<ExampleEmail />, { plainText: true })
console.log({ text })
```
